src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/actions/ModifyCacheMaxSizeAction.java

```diff
@@ -34,49 +33,44 @@
  * deciders to implement actions like increasing the cache's size. Presently, it acts on field data
  * cache and shard request cache.
  */
-
-// TODO: Split the cache action into separate actions for different caches.
-
 public class ModifyCacheMaxSizeAction extends SuppressibleAction {
   private static final Logger LOG = LogManager.getLogger(ModifyCacheMaxSizeAction.class);
-  public static final String NAME = "modifyCacheCapacity";
-  public static final long COOL_OFF_PERIOD_IN_MILLIS = 300 * 1_000;
+  public static final String NAME = "ModifyCacheCapacity";
 
   private final NodeKey esNode;
   private final ResourceEnum cacheType;
-  private final NodeConfigCache nodeConfigCache;
-  private final double cacheSizeUpperBound;
+  private final AppContext appContext;
 
-  private Long currentCacheMaxSizeInBytes;
-  private Long heapMaxSizeInBytes;
-  private long cacheUpperBoundInBytes;
-  private long desiredCacheMaxSizeInBytes;
-
-  private Map<ResourceEnum, Long> stepSizeInBytes = new HashMap<>();
+  private final long desiredCacheMaxSizeInBytes;
+  private final long currentCacheMaxSizeInBytes;
+  private final long coolOffPeriodInMillis;
+  private final boolean canUpdate;
 
   public ModifyCacheMaxSizeAction(
       final NodeKey esNode,
       final ResourceEnum cacheType,
-      final NodeConfigCache nodeConfigCache,
-      final double cacheSizeUpperBound,
-      final boolean increase,
-      final AppContext appContext) {
-    // TODO: Add lower bound for caches
-    // TODO: Address cache scaling down  when JVM decider is available
-
+      final AppContext appContext,
+      final long desiredCacheMaxSizeInBytes,
```

Comment:
long primitive itself is immutable. We don't need to explicitly specify it as final

Reply:
The intent of adding final to these values to make them immutable. explicitly declare it as final so we won’t try to make any changes to that variable within the constructor. 

Based on our discussion. Removing final with setter's and retaining them in the constructors.